torch_xla/csrc/xla_graph_executor.h

```diff
@@ -230,6 +271,10 @@ class XLAGraphExecutor : public torch::lazy::LazyGraphExecutor {
     std::unordered_map<torch::lazy::hash_t, std::vector<xla::Shape>,
                        torch::lazy::HashReducer>
         hash_to_output_shape_map;
+    // Keep track of sharded tensors. This assumes that each XLATensor::Data
+    // element is alive throughout the lifetime of DeviceContextArena.
+    std::unordered_map<XLATensor::Data*, size_t> sharded_tensors_map;
+    std::vector<std::shared_ptr<XLATensorImpl>> sharded_tensor_impl_vec;
```

Comment:
Why should we keep `sharded_tensor_impl_vec` here?

Reply:
Lol, thsi is probably the left over of my previous experiments, I was experimenting with 3 different options to fix the bug before settling with the one. I will clean up and ping you.

Reply:
Removed. Thanks @alanwaketan